src/System.Management.Automation/engine/Attributes.cs

```diff
@@ -2163,6 +2204,42 @@ protected override void Validate(object arguments, EngineIntrinsics engineIntrin
         }
     }
 
+    /// <summary>
+    /// Validates that the parameters's argument is not null, is not an empty string, and is
+    /// not an empty collection. If argument is a collection, each argument is verified.
+    /// </summary>
+    [AttributeUsage(AttributeTargets.Field | AttributeTargets.Property)]
+    public sealed class ValidateNotNullOrEmptyAttribute : ValidateNotNullOr
+    {
+        /// <summary>
+        /// Validates that the parameters's argument is not null, is not an empty string, and is
+        /// not an empty collection. If argument is a collection, each argument is verified.
+        /// </summary>
+        public ValidateNotNullOrEmptyAttribute()
+            : base(false)
```

Comment:
Please be explicit on the parameter name for constant arguments. It helps the code to be more readable.

```suggestion
            : base(checkWhiteSpace: false)
```